cmd/kubeadm/app/util/staticpod/utils.go

```diff
@@ -69,8 +70,10 @@ func ComponentPod(container v1.Container, volumes map[string]v1.Volume, annotati
 			Labels:      map[string]string{"component": container.Name, "tier": kubeadmconstants.ControlPlaneTier},
 			Annotations: annotations,
 		},
+
 		Spec: v1.PodSpec{
 			Containers:        []v1.Container{container},
+			Priority:          &priority,
```

Comment:
we already have a priorityclass set on the pods. unless, i am missing something this priority field is not needed.

Reply:
see the linked bug, without it Graceful Node Shutdown doesn't work properly
In https://github.com/kubernetes/kubernetes/pull/89970 priority was explicitly set with the comment `adds numeric priority to ensure relative preemption works properly`. My working theory is that kubelet see the static pod without the priority (as it is on disk), but other components that use the API see it with the priority

This PR is more of a workaround than a fix, but my manual tests show that setting the priority make a difference

Reply:
i'd prefer if we don't include similar workarounds unless it's necessary. from my understaning the class name should work as is, so perhaps if there is a bug it must be fixed instead.

cc @liggitt 
https://github.com/kubernetes/kubernetes/issues/113940#issuecomment-1341050952

Reply:
I'm pretty sure static pods use the values from the manifest and not the API

Reply:
ok, still not clear to me if this is a bug. sounds like the static pods just skip asking the api :/
is 2000001000 just an arbitrary number or is there a constant for it somewhere?

we should at least have a comment on top of the int() explaining why this is done.

Reply:
2000001000 is the numeric priority that corresponds to `system-node-critical`:

https://github.com/kubernetes/kubernetes/blob/09d7286eb570c8ab89c3008efd6afcf590a94a50/pkg/apis/scheduling/v1/helpers.go#L31-L36

Reply:
ok, please update with a comment what the int corresponds to, @champtar